controllers/kubeadmbootstrapconfig_controller_test.go

```diff
@@ -71,21 +71,32 @@ func (c *myClient) Status() client.StatusWriter {
 
 func TestSuccessfulReconcileShouldNotRequeue(t *testing.T) {
 	machineKind := v1alpha2.SchemeGroupVersion.WithKind("Machine").String()
-
-	myclient := &myClient{
-		db: map[string]runtime.Object{
-			"ns/cfg": &kubeadmv1alpha1.KubeadmBootstrapConfig{
-				ObjectMeta: metav1.ObjectMeta{
-					Namespace: "ns",
-					Name:      "cfg",
-					OwnerReferences: []metav1.OwnerReference{
-						{
-							Kind: machineKind,
-						},
+	objects := map[string]runtime.Object{
```

Comment:
The two tests look really similar right now, but the unit test gives us full control over the client. We can return errors and such that we cannot force through the k8s client. We also don't have to work schemes and such using the unit test. More control, but exercises fewer pieces.